src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.RazorProviders.cs

```diff
@@ -53,12 +53,13 @@ public partial class RazorSourceGenerator
                 ? false
                 : CSharpCompilation.Create("components", references: minimalReferences).HasAddComponentParameter();
 
-            var razorConfiguration = new RazorConfiguration(razorLanguageVersion, configurationName ?? "default", Extensions: [], UseConsolidatedMvcViews: true, SuppressAddComponentParameter: !isComponentParameterSupported);
 
             // We use the new tokenizer only when requested for now.
             var useRoslynTokenizer = parseOptions.Features.TryGetValue("use-roslyn-tokenizer", out var useRoslynTokenizerValue)
                                     && string.Equals(useRoslynTokenizerValue, "true", StringComparison.OrdinalIgnoreCase);
 
+            var razorConfiguration = new RazorConfiguration(razorLanguageVersion, configurationName ?? "default", Extensions: [], UseConsolidatedMvcViews: true, SuppressAddComponentParameter: !isComponentParameterSupported);
+
             var razorSourceGenerationOptions = new RazorSourceGenerationOptions()
```

Comment:
Thiss constructor ends up with an object where `options.UseRoslynTokenizer != options.Configuration.UseRoslyntokenizer`. Is there a place where we could add validation to catch this dissagreement? Or is it a valid setup and I'm missing it. Could `RazorSourceGenerationOptions` just have `UseRoslynTokenizer` be a passthrough to the property on `Configuration`?

Reply:
RazorConfiguration is currently not ideal IMO. It's why I logged https://github.com/dotnet/razor/issues/11182. Tooling passes a RazorConfiguration to the compiler when configuring the project engine, but the compiler doesn't use all of the configuration, so we also have to read properties off that configuration again, and do more config in the `configure` lambda. Ideally we'd just give the compiler some type it needed, and it would go from there.

So yeah, `UseRoslynTokenizer` could be passed in here, and in fact thats why I moved it, but then I realised it wasn't necessary and forgot to move it back.